packages/effect/src/internal/stream.ts

```diff
@@ -8045,3 +8045,17 @@ export const encodeText = <E, R>(self: Stream.Stream<string, E, R>): Stream.Stre
     const encoder = new TextEncoder()
     return map(self, (s) => encoder.encode(s))
   })
+
+/** @internal */
+export const fromEventListener = (
+  target: EventTarget,
+  type: string,
+  options?: boolean | Omit<AddEventListenerOptions, "signal">
+): Stream.Stream<Event> =>
+  _async<Event>((emit) => {
+    function cb(e: Event) {
+      emit.single(e)
+    }
+    target.addEventListener(type, cb, options)
```

Comment:
Might be a good idea to batch the events:

```suggestion
    let batch: Array<Event> = []
    let taskRunning = false
    function cb(e: Event) {
      batch.push(e)
      if (!taskRunning) {
        taskRunning = true
        queueMicrotask(() => {
          const events = batch
          batch = []
          taskRunning = false
          emit.chunk(Chunk.unsafeFromArray(events))
        })
      }
    }
    target.addEventListener(type, cb, options)
```

Reply:
Hmm. Testing this, it doesn't seem to be actually batching the events. It's just emitting a new chunk for each event.

Reply:
Do you think it would make sense to use a `window:Duration` for batching?

Reply:
No I think this is fine :)